python/ray/train/_internal/state/schema.py

```diff
@@ -1,10 +1,25 @@
+from enum import Enum
 from typing import List, Optional
 
 from ray._private.pydantic_compat import BaseModel, Field
 from ray.dashboard.modules.job.pydantic_models import JobDetails
 from ray.util.annotations import DeveloperAPI
 
 
+@DeveloperAPI
+class RunStatusEnum(str, Enum):
+    STARTED = "STARTED"
+    FINISHED = "FINISHED"
+    ERRORED = "ERRORED"
+    ABORTED = "ABORTED"
```

Comment:
What is the difference between ERRORED and ABORTED?

Reply:
We only mark it as `ERRORED` when Trainer detected an error and actively report it to `StateActor`.

We mark it as `ABORTED` when the TrainRun abnormally failed (e.g. due to node failure) and we passively set it in `TrainHead`.